manifests/config.pp

```diff
@@ -74,13 +26,15 @@
   }
 
   if $::osfamily == 'RedHat' {
-    file { '/etc/oddjobd.conf.d/oddjobd-mkhomedir.conf':
-      ensure  => 'file',
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0644',
-      content => template('winbind/oddjobd-mkhomedir.conf.erb'),
-      notify  => Service[['oddjobd', 'winbind',]],
+    if ($winbind::manage_oddjob_service == true) {
```

Comment:
This file needs managing regardless of if the service is managed so this `if` should not be here.